### DISCUSSION TITLE
Workload with id maui is not supported on this platform

### DISCUSSION THREAD

**Discussion by ldl01031** [dotnet/maui#2075] [General]
Not sure where to go next. Ideally, I could clean things up to start with a clean slate, but I have no idea how to do that.

I'm on Windows 10, 64-bit, running VS 2022 version 17 preview 3.0

Running: dotnet workload install maui give the following error:
Workload with id maui is not supported on this platform.

I've tried a hundred things and seem 'closer' but the error above is a show stopper.

Is there a simple way (or a checklist) to uninstall everything to do with maui? So I can start fresh? Or, other ideas?

**Comment by blazorin** [comment]
Hello, if you didn't try [Redth's maui-check tool](https://github.com/Redth/dotnet-maui-check) before, try it:

To install:
`dotnet tool install -g Redth.Net.Maui.Check`
To run:
`maui-check`

It will try to set up the Maui workloads apart from checking you have the rest of the requirements, if you have trouble with it, there are other solutions I can tell you.

**Reply by ldl01031** [reply]
I've run that more times than I can count :)

I had trouble with that utility in the beginning because it kept install .net preview, then saying it wasn't installed, do I want to fix it? (repeat)
I finally got that resolved by manually uninstalling/reinstalling that preview.

Now maui-check says everything is good, but it is not. One example is the error above when running "dotnet workload install maui"

**Reply by ldl01031** [reply]
Here is an example of the errors when I open a maui project in vs 2022 (and the project was created by a command line tool):

![Screenshot 2021-08-12 181955](https://user-images.githubusercontent.com/37428199/129291454-bc6cc747-8dca-4ed6-8478-7b4a5e7d706d.jpg)

**Reply by GeraudFabien** [reply]
I compiled by removing net6.0-ios and net6.0-maccatalyst from the csproj. But i still can't run it:
![image](https://user-images.githubusercontent.com/13610689/129318050-de2c3340-2ba6-4a89-aaaf-f571a5ea424e.png)

Also did you installed all the needed worload [before using maui check](https://docs.microsoft.com/en-us/dotnet/maui/get-started/installation):
> To create .NET MAUI apps in Visual Studio, you'll also need the latest Visual Studio 2022 Preview with the following workloads installed:
> 
> Mobile development with .NET
> Universal Windows Platform development
> Desktop development with C++
> .NET Desktop Development
> ASP.NET and web development (required for Blazor Desktop and the BlazorWebView control)

**Reply by ldl01031** [reply]
Yep, I did everything. I think I 'might' have skipped over "dotnet workload install maui" the first time around. Now, when I run that I get the "Workload with id maui is not supported on this platform" error. I've uninstalled/reinstalled everything I know of, but am still stopped by this error.

**Reply by taozuhong** [reply]
Windows: Windows Machine + Framework(net6.0-windows****)
Android: Android Emulator + Framework(net6.0-android)

**Comment by guide2wsl** [comment]
I also have this same problem, running Windows 11 on ARM64:
```powershell
dotnet --version
6.0.100-preview.7.21379.14

dotnet workload install maui
Workload with id maui is not supported on this platform.

# plan B also fails:
dotnet tool install -g Redth.Net.Maui.Check
C:\Users\guide2wsl\AppData\Local\Temp\sgvqb0lg.gfa\restore.csproj : error NU1100: Unable to resolve 'redth.net.maui.check (>= 0.0.0)' for 'net6.0'.
C:\Users\guide2wsl\AppData\Local\Temp\sgvqb0lg.gfa\restore.csproj : error NU1100: Unable to resolve 'redth.net.maui.check (>= 0.0.0)' for 'net6.0/any'.
The tool package could not be restored.
```

I'm just trying to follow https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app?pivots=dotnet-cli without much success at the moment.  Admittedly, the page says, "The documentation for .NET Multi-app Platform UI (.NET MAUI) is under construction." so maybe I should wait 🙂

**Reply by blazorin** [reply]
> I also have this same problem, running Windows 11 on ARM64:
> 
> ```powershell
> dotnet --version
> 6.0.100-preview.7.21379.14
> 
> dotnet workload install maui
> Workload with id maui is not supported on this platform.
> 
> # plan B also fails:
> dotnet tool install -g Redth.Net.Maui.Check
> C:\Users\guide2wsl\AppData\Local\Temp\sgvqb0lg.gfa\restore.csproj : error NU1100: Unable to resolve 'redth.net.maui.check (>= 0.0.0)' for 'net6.0'.
> C:\Users\guide2wsl\AppData\Local\Temp\sgvqb0lg.gfa\restore.csproj : error NU1100: Unable to resolve 'redth.net.maui.check (>= 0.0.0)' for 'net6.0/any'.
> The tool package could not be restored.
> ```
> 
> I'm just trying to follow https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app?pivots=dotnet-cli without much success at the moment. Admittedly, the page says, "The documentation for .NET Multi-app Platform UI (.NET MAUI) is under construction." so maybe I should wait 🙂

Try now with .NET 6 RC1, it should use different.NET paths and may solve that. Let me know.

**Reply by guide2wsl** [reply]
Nope :(

```powershell
> dotnet --version
6.0.100-rc.1.21458.32

> dotnet workload install maui
Workload ID maui isn't supported on this platform.

> (Get-CimInstance Win32_operatingsystem).OSArchitecture
ARM 64-bit Processor
```
